thunder/torch/__init__.py

```diff
@@ -187,6 +187,68 @@ def is_cuda(a: TensorLike, /) -> bool:
 register_method("size", size)
 
 
+@torchsymbol(torch.Tensor.type, is_method=True)
+def type(a: TensorLike, dtype: None | str | dtypeLike = None, non_blocking: bool = False, /) -> str | TensorLike:
+    utils.check(
+        not non_blocking,
+        lambda: f"type(): `non_blocking==True` is currently not supported.",
+        exception_type=NotImplementedError,
+    )
+
+    DTYPE_STR = {
```

Comment:
This is to interesting. This PR is absolutely correct that PyTorch does support putting in a string for `dtype`, and that behavior is used by some programs, but not many. 

I wonder if we should just throw a `NotImplementedError` if a string is passed as the `dtype` argument for now, and see if we get a request to support it

Reply:
Yeah, that is also possible and reasonable. However as per the issue tag, it seems that it is trying to implement Megatron? When I do a quick search about the usage of `type()` it is something like the [following](https://arc.net/l/quote/fmtstkye):
```python
                    if param.type() in ['torch.cuda.HalfTensor', 'torch.cuda.BFloat16Tensor']:
                        float16_params_this_group.append(param)
```
What do you think? When observing the use case, it seems like it is better to support string.

Reply:
Yeah, you have a point. I think it's terrible code compared to `param.is_cuda and param.dtype in (torch.float16, torch.bfloat16)`  but I guess our mission is not to run only good code. :)

Reply:
Awesome to look at the actual use case -- thanks @k223kim